internal/api/external_naver_test.go

```diff
@@ -0,0 +1,184 @@
+	response := naverResponse
+	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, response)
+	defer server.Close()
+
+	u := performAuthorization(ts, "naver", code, "")
+
+	assertAuthorizationSuccess(ts, u, tokenCount, userCount, "naver@example.com", "Naver Test", "123", "http://example.com/avatar")
+}
+
+func (ts *ExternalTestSuite) TestInviteTokenExternalNaverSuccessWhenMatchingToken() {
+	// name and avatar should be populated from Naver API
+	ts.createUser("123", "naver@example.com", "", "", "invite_token")
+
+	tokenCount, userCount := 0, 0
+	code := "authcode"
+	response := naverResponse
+	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, response)
+	defer server.Close()
+
+	u := performAuthorization(ts, "naver", code, "invite_token")
+
+	assertAuthorizationSuccess(ts, u, tokenCount, userCount, "naver@example.com", "Naver Test", "123", "http://example.com/avatar")
+}
+
+func (ts *ExternalTestSuite) TestInviteTokenExternalNaverErrorWhenNoMatchingToken() {
+	tokenCount, userCount := 0, 0
+	code := "authcode"
+	response := naverResponse
+	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, response)
+	defer server.Close()
+
+	w := performAuthorizationRequest(ts, "naver", "invite_token")
+	ts.Require().Equal(http.StatusNotFound, w.Code)
+}
+
+func (ts *ExternalTestSuite) TestInviteTokenExternalNaverErrorWhenWrongToken() {
+	ts.createUser("123", "naver@example.com", "", "", "invite_token")
+
+	tokenCount, userCount := 0, 0
+	code := "authcode"
+	response := naverResponse
+	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, response)
+	defer server.Close()
+
+	w := performAuthorizationRequest(ts, "naver", "wrong_token")
+	ts.Require().Equal(http.StatusNotFound, w.Code)
+}
+
+func (ts *ExternalTestSuite) TestInviteTokenExternalNaverErrorWhenEmailDoesntMatch() {
+	ts.createUser("123", "naver@example.com", "", "", "invite_token")
+
+	tokenCount, userCount := 0, 0
+	code := "authcode"
+	response := naverResponseAnotherEmail
+	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, response)
+	defer server.Close()
+
+	u := performAuthorization(ts, "naver", code, "invite_token")
+
+	assertAuthorizationFailure(ts, u, "Invited email does not match emails from external provider", "invalid_request", "")
+}
+
+// func (ts *ExternalTestSuite) TestSignupExternalNaverErrorWhenVerifiedFalse() {
+// 	tokenCount, userCount := 0, 0
+// 	code := "authcode"
+// 	emails := `[{"email":"naver@example.com", "primary": true, "verified": false}]`
+// 	server := NaverTestSignupSetup(ts, &tokenCount, &userCount, code, emails)
+// 	defer server.Close()
+
+// 	u := performAuthorization(ts, "naver", code, "")
+
+// 	v, err := url.ParseQuery(u.Fragment)
+// 	ts.Require().NoError(err)
+// 	ts.Equal("unauthorized_client", v.Get("error"))
+// 	ts.Equal("401", v.Get("error_code"))
+// 	ts.Equal("Unverified email with naver", v.Get("error_description"))
+// 	assertAuthorizationFailure(ts, u, "", "", "")
+// }
```

Comment:
Please remove the commented-out code

Reply:
Oops, I just commented it out.

I'll remove it.